doozer/doozerlib/distgit.py

```diff
@@ -1394,10 +1394,66 @@ def _clean_repos(self, dfp):
                 if changed:
                     dfp.add_lines_at(entry, "RUN " + new_value, replace=True)
 
-    def _mapped_image_from_member(self, image, original_parents, count):
+    def _resolve_parent(self, original_parent, dfp):
+        try:
+            self.logger.debug('Retrieving image info for image %s', original_parent)
+            cmd = f'oc image info {original_parent} -o json'
+            out, _ = exectools.cmd_assert(cmd, retries=3)
+            labels = json.loads(out)['config']['config']['Labels']
+
+            # Get the exact build NVR
+            build_nvr = f'{labels["com.redhat.component"]}-{labels["version"]}-{labels["release"]}'
+
+            # Query Brew for build info
+            self.logger.debug('Retrieving info for Brew build %s', build_nvr)
+            with self.runtime.shared_koji_client_session() as koji_api:
+                if not koji_api.logged_in:
+                    koji_api.gssapi_login()
+                build = koji_api.getBuild(build_nvr, strict=True)
+
+            # Get the pullspec for the upstream equivalent
+            upstream_equivalent_pullspec = build['extra']['image']['index']['pull'][1]
```

Comment:
nitpick, we call this "upstream_equivalent_pullspec", imo it would be simpler to call it "downstream_image_pullspec"

Reply:
This is really the pullspec of the image that upstream wants to use, since it is deduced directly from the upstream Dockerfile. Naming it `downstream` would sound confusing to me